src/main/java/com/redhat/pantheon/conf/LocalFileManagementService.java

```diff
@@ -38,54 +40,58 @@ public void activate() throws IOException {
     }
 
     public void initializeTemplateDirectories() throws IOException {
-        Enumeration<URL> urls = FrameworkUtil.getBundle(LocalFileManagementService.class)
-                .findEntries("apps/pantheon/templates/haml/html5", "*", false);
+        Optional<Enumeration<URL>> urls = Optional.ofNullable(FrameworkUtil.getBundle(LocalFileManagementService.class)
+                .findEntries("apps/pantheon/templates/haml/html5", "*", false));
         Path p = Files.createTempDirectory("templates");
 
         log.info("Initializing template directories at " + p.toString());
 
-        templateDirectory = Optional.of(p.toFile());
-
-        while (urls.hasMoreElements()) {
-            URL url = urls.nextElement();
-            String filename = url.toString();
-            filename = filename.substring(filename.lastIndexOf("/") + 1);
-
-            File f = new File(templateDirectory.get(), filename);
-            f.deleteOnExit();
-
-            InputStream is = url.openConnection().getInputStream();
-
-            FileUtils.copyInputStreamToFile(is, f);
-        }
+        urls.ifPresent(a -> templateDirectory = Optional.of(p.toFile()));
```

Comment:
@carlosmunoz This is what I'm talking about in the other comment.

Reply:
I think a traditional if statement makes this more readable. The fact that `a` is not consumed is a bit of a distraction.

Reply:
Ok, I understand. I put the if-statement back (using isPresent instead of orElse).